fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAnalyzerRegistryImpl.java

```diff
@@ -88,29 +89,63 @@ private LuceneAnalyzerRegistryImpl() {
-            Pair<AnalyzerChooser, AnalyzerChooser> perFieldAnalyzerChooserPair = getAnalyzerChooser(index, analyzerName, type);
-            indexAnalyzerChooserPerFieldOverride.put(fieldName, perFieldAnalyzerChooserPair.getLeft());
-            queryAnalyzerChooserPerFieldOverride.put(fieldName, perFieldAnalyzerChooserPair.getRight());
+        if (analyzerPerFieldName != null) {
+            LuceneIndexOptions.parseKeyValuePairOptionValue(analyzerPerFieldName).forEach((fieldName, analyzerName) -> {
+                Pair<AnalyzerChooser, AnalyzerChooser> perFieldAnalyzerChooserPair = getAnalyzerChooser(index, analyzerName, type);
+                indexAnalyzerChooserPerFieldOverride.put(fieldName, perFieldAnalyzerChooserPair.getLeft());
+                queryAnalyzerChooserPerFieldOverride.put(fieldName, perFieldAnalyzerChooserPair.getRight());
+            });
+        }
+
+        auxiliaryFieldInfo.forEach((fieldName, fieldInfo) -> {
+            addPerFieldAnalyzerIfNecessary(indexAnalyzerChooserPerFieldOverride, fieldName, fieldInfo, index);
+            addPerFieldAnalyzerIfNecessary(queryAnalyzerChooserPerFieldOverride, fieldName, fieldInfo, index);
         });
 
         return new LuceneAnalyzerCombinationProvider(defaultAnalyzerChooserPair.getLeft(), defaultAnalyzerChooserPair.getRight(),
                 indexAnalyzerChooserPerFieldOverride, queryAnalyzerChooserPerFieldOverride);
     }
 
+    private void addPerFieldAnalyzerIfNecessary(@Nonnull final Map<String, AnalyzerChooser> chooserPerFieldOverride,
+                                                @Nonnull final String fieldName,
+                                                @Nonnull final LuceneIndexExpressions.DocumentFieldDerivation fieldInfo,
+                                                @Nonnull final Index index) {
+        if (chooserPerFieldOverride.containsKey(fieldName)) {
+            // do not override already chosen field analyzer.
+            return;
+        }
+        if (isEligibleForNoOpAnalyzer(fieldInfo)) {
+            if (registry.isEmpty()
+                    || registry.get(LuceneAnalyzerType.FULL_TEXT) == null
+                    || registry.get(LuceneAnalyzerType.FULL_TEXT).get(ExactTokenAnalyzerFactory.NAME) == null) {
+                throw new MetaDataException("could not retrieve analyzer",
+                        LuceneLogMessageKeys.ANALYZER_NAME, ExactTokenAnalyzerFactory.NAME,
+                        LuceneLogMessageKeys.ANALYZER_TYPE, LuceneAnalyzerType.FULL_TEXT);
+            }
+            chooserPerFieldOverride.put(fieldName, registry.get(LuceneAnalyzerType.FULL_TEXT).get(ExactTokenAnalyzerFactory.NAME).getIndexAnalyzerChooser(index));
+        }
+    }
+
+    private static boolean isEligibleForNoOpAnalyzer(@Nonnull final LuceneIndexExpressions.DocumentFieldDerivation fieldInfo) {
+        return fieldInfo.isSorted() || fieldInfo.isStored();
```

Comment:
Rather than looking at the `sorted` and `stored` of the field, should it look at the `fieldInfo`'s type instead? In particular, I'd think you'd want to use the exact analyzer unless the type is `TEXT` (as only `TEXT` fields are going to actually bo tokenized in the index)

Reply:
That's interesting, I changed the check to:
```java
return fieldInfo.getType() != LuceneIndexExpressions.DocumentFieldType.TEXT;
```
I am a bit worried about `STRING` type though. and more generally whether this no-op analyser can substitute, correctly, the `StandardAnalyser` (for cases where the analyser is not defined, and the fields are _not_ `stored` or `sorted`).

Reply:
I think for `STRING` type fields, we want the no-op analyzer, because the field shouldn't have been analyzed before being indexed...if it is indexed, somehow. At least, that's my understanding

Reply:
Ideally the PR description and summary would be changed to reflect this new, simpler determining factor.